<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [154]:
# Membuat dataframes dari dictionary
import pandas as pd

corona = {'Jml Terinfeksi': ['10', '230', '2900', '10000', '17000'], 
              'bulan': ['January', 'Februari', 'Maret', 'April', 'Mei'], 
               'tahun': ['2020', '2020', '2020', '2020', '2020']} 
  

displayData = pd.DataFrame(corona)
print(displayData)

  Jml Terinfeksi     bulan tahun
0             10   January  2020
1            230  Februari  2020
2           2900     Maret  2020
3          10000     April  2020
4          17000       Mei  2020


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [155]:
# Membaca data dan memasukannya kedalam suatu variabel
import pandas as pd

csv = pd.read_csv('diamonds.csv')
print(csv)


       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 

A. perbedaan Series dengan list dan dictionary
- list : bisa menggunakan tipe data seperti integer, string, dan lain sebagainya dan tipe data object tidak harus sama.
- Dictionary : bisa menggunakan tipe data seperti integer, string, dan lain sebagainya dan tipe data object tidak harus sama tapi membutuhkan key yang bertipe data string untuk memanggil objek dalam dictionary.
- Series : bisa menggunakan tipe data seperti integer, string, dan lain sebagainya tapi tipe data dari objek series ini harus seragam.

B. perbedaan dataframe dan series
- series : Series adalah suatu objek satu dimensi yang dapat menyimpan berbagai jenis tipe data seperti integer, string, dan lain sebagainya.
- Data Frame : DataFrame adalah suatu objek 2 dimensi tempat menyimpan data dengan lebih terstruktur.

In [156]:
# code here
import pandas as pd

data = [[1, 2, 3, 4, 5],[6,7,8,9,10]]
print(pd.Series(data))

0     [1, 2, 3, 4, 5]
1    [6, 7, 8, 9, 10]
dtype: object


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [157]:
# code hereimport pandas as pd
import pandas as pd

df = pd.read_csv('diamonds.csv')
def price():
    ac = df.head(3)
    return ac["price"].iloc[0]
def color():
    ab = df.tail(3)
    return ab["color"].iloc[-1]

print(df.head(3))
print("\n")
print(price())
print("\n")
print(df.tail(3))
print("\n")
print(color())


   Unnamed: 0  carat      cut color clarity  depth  table  price     x     y  \
0           1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98   
1           2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84   
2           3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07   

      z  
0  2.43  
1  2.31  
2  2.31  


326


       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64  


D


<h2>Soal 5: Deskriptif Statistik</h2>



Statistik deskriptif adalah metode dari pengorganisasian, penjumlahan, dan penyajian data dalam sebuah cara yang nyaman dan informatif, termasuk teknik grafik dan teknik penghitungan.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [158]:
# code here
# import statistics
import pandas as pd

df = pd.read_csv('diamonds.csv')
descData = df.describe()
def meanPrice():
    return descData["price"]["mean"]
def deviasiDepth():
    return descData["depth"]["std"]
def maximumCarat():
    return descData["carat"]["max"]


print(descData)
print("\njawaban\n")
print("nilai rata-rata dari column price : ", meanPrice())
print("nilai standar deviasi dari column depth : ", deviasiDepth())
print("nilai maximum dari column carat :", maximumCarat())

         Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000     

Jawab:
- nilai rata-rata dari column price : 3932.799722
- nilai standar deviasi dari column depth : 1.432621
- nilai maximum dari column carat : 5.010000